lib/merchant_repository.rb

```diff
@@ -0,0 +1,41 @@
+class MerchantRepository
+  # This is the MerchantRepository Class
+  # frozen_string_literal: true.
+  attr_reader :merchants
+
+  def initialize
+    @merchants = []
+  end
+
+  def create(data)
+    data[:id] ||= 1 if merchants == []
```

Comment:
The parent repo currently does not account for an empty repo. Not necessary if we initialize with objects but this line will need to be added if we change the implementation to mirror yours.